pkg/alerter/alerter.go

```diff
@@ -0,0 +1,79 @@
+package alerter
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	"github.com/sirupsen/logrus"
+)
+
+type Alerter struct {
+	sync.Mutex
+	Alerts   Alerts
+	AlertsCh chan Alerts
+}
+
+func NewAlerter() *Alerter {
+	return &Alerter{
+		Alerts:   make(Alerts),
+		AlertsCh: make(chan Alerts),
+	}
+}
+
+func (a *Alerter) Start() {
+	go func() {
+		defer close(a.AlertsCh)
+		for {
+			alert := <-a.AlertsCh
+			a.mergeAlerts(alert)
+		}
+	}()
+}
+
+func (a *Alerter) Alert(key string, alert Alert) {
+	a.AlertsCh <- Alerts{
+		key: []Alert{alert},
+	}
+}
+
+func (a *Alerter) mergeAlerts(alert Alerts) {
+	a.Lock()
+	defer a.Unlock()
+	for k, v := range alert {
+		if val, ok := a.Alerts[k]; ok {
+			a.Alerts[k] = append(val, v...)
+		} else {
+			a.Alerts[k] = v
+		}
+	}
+}
+
+func (a *Alerter) IgnoreResources(resources []resource.Resource) []resource.Resource {
```

Comment:
I remembered our last talk where we described the "ignoring part" being somewhat similar to the one we did for the driftignore. Since we did it right before the analysis starts, I put the IgnoreResources right before the analysis too in the `driftctl.go` file.

Would love @moadibfr's feedback on this one.

Reply:
I agree with Elie on the name and on the place.
In my PR on ignore I had the same problem added argument to Analyzer to pass a Filterer. I'm not sure this is ideal either

Reply:
Lets close this one and continue discussion there : https://github.com/cloudskiff/driftctl/pull/28/files#discussion_r545750961